packages/astro/src/server/clerk-middleware.ts

```diff
@@ -234,6 +237,27 @@ Check if signInUrl is missing from your configuration or if it is not an absolut
    PUBLIC_CLERK_SIGN_IN_URL='SOME_URL'
    PUBLIC_CLERK_IS_SATELLITE='true'`;
 
+/**
+ * Prevents infinite redirects in Netlify's functions
+ * by adding a cache bust parameter to the original redirect URL. This ensures Netlify
+ * doesn't serve a cached response during the authentication flow.
+ */
+function handleNetlifyCacheInDevInstance(locationHeader: string, requestState: RequestState) {
+  // Only run on Netlify environment
+  // eslint-disable-next-line turbo/no-undeclared-env-vars
+  if (!process.env.NETLIFY) {
```

Comment:
Do we have a way to check if Clerk is running in dev instance? This redirect loop issue happens only in Netlify and Clerk development instance

Reply:
You can use these: https://github.com/clerk/javascript/blob/52a24eba0d71635cea3bc2f3380aa698907b63a0/packages/shared/src/keys.ts#L97-L111

Reply:
Perfect, thanks @LekoArts!